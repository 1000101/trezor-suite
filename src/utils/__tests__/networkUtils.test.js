import * as networkUtils from '../networkUtils';

describe('network utils', () => {
    beforeEach(() => {
        fetch.resetMocks();
    });

    it('httpRequest json ok response json', async () => {
        fetch.mockResponse(JSON.stringify({ test_json: '12345' }));
        const result = await networkUtils.httpRequest('/http-request-test-response-json/', 'json');
        expect(result).toMatchSnapshot();
    });

    it('httpRequest json ok response text', async () => {
        fetch.mockResponse(JSON.stringify({ test_json: '12345' }));
        const result = networkUtils.httpRequest('/http-request-test-response-text/', 'text');
        expect(result).toMatchSnapshot();
    });

    it('httpRequest json ok response binary', async () => {
        fetch.mockResponse(1);
        const result = networkUtils.httpRequest('/http-request-test-response-binary/', 'binary');
        expect(result).toMatchSnapshot();
    });

    it('httpRequest error (500)', async () => {
        fetch.mockResponse(JSON.stringify({ test_json: '12345' }), { status: 500 });
        try {
            await networkUtils.httpRequest('/test/', 'json');
        } catch (err) {
            expect(err).toMatchSnapshot();
        }
    });

    it('JSONRequest', async () => {
        fetch.mockResponse(JSON.stringify({ test_json: '01234' }));
        const result = await networkUtils.JSONRequest('/test/');
        expect(result).toMatchSnapshot();
    });

    it('JSONRequest error', async () => {
        fetch.mockResponse(JSON.stringify({ test_json: '12345' }), { status: 500 });
        try {
            await networkUtils.JSONRequest('/test/', 'json');
        } catch (err) {
            expect(err).toMatchSnapshot();
        }
    });
});
