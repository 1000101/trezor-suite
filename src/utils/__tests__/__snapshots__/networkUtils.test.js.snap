// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`network utils JSONRequest 1`] = `
Object {
  "test_json": "01234",
}
`;

exports[`network utils JSONRequest error 1`] = `[Error: jsonRequest error: [object Object]]`;

exports[`network utils httpRequest error (500) 1`] = `[Error: /test/ Internal Server Error]`;

exports[`network utils httpRequest json ok response binary 1`] = `Promise {}`;

exports[`network utils httpRequest json ok response json 1`] = `
Object {
  "test_json": "12345",
}
`;

exports[`network utils httpRequest json ok response text 1`] = `Promise {}`;
